### DISCUSSION TITLE
[Kavita] Request for Pagination and "Sync Read Progress" Support for Cantook Client in Kavita

### DISCUSSION THREAD

**Discussion by ririnto** [Kareadita/Kavita#3379] [Ideas]
### Idea Description

The Cantook client currently does not support **Pagination** and **"Sync Read Progress"** features when integrated with Kavita.

Komga supports these features with Cantook, as described in their [documentation](https://komga.org/docs/guides/opds/). Adding similar functionality to Kavita would greatly enhance usability and user experience.

Please consider implementing these features. Thank you.

### Idea Category

API

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by ririnto** [comment]
This request is very similar to **#2590**, but it emphasizes the need for more robust Pagination functionality.

**Comment by majora2007** [comment]
What you are asking for is OPDS2.0 Support. I took a look into it and am not convinced, which I have shared with Panels team as well. From my investigation, there is not much benefit for the users and there is only the official client implementing it. I doubt other applications will implement. 

As of now, this is not a priority for me to implement. Kavita supports Pagination and PS extension for OPDS 1 feeds and OPDS 1 has many clients. I'm more of the opinion to expand OPDS 1 with features that commonly come up in Panels server for a richer experience, than build a new spec that has nothing to do with the type of medium we deal in.

**Reply by ririnto** [reply]
I understand that implementing this feature could be challenging due to the limited number of expected users. Currently, the system works fine on a single device, except for an issue with pagination—for instance, only the first page is retrieved when querying books from the catalog. While the "Sync Read Progress" feature would be nice to have, it’s not a critical need at the moment.

Since I was looking for a reader that supports both Android and iOS, I decided to use this client. As a Java developer, I might find this issue challenging to resolve, but would it be alright if I try to improve it when I have the time? If so, could you advise me on which parts I should prioritize first?

Thank you for your understanding, as English is not my first language, and there might be some awkward expressions.

**Reply by majora2007** [reply]
Kavita already offers Pagination support via OPDS, I can't speak to Cantook as I've tried it once and it wasn't a good experience. If they support OPDS 1.2, there should be no issue. It sounds like they don't follow the spec for Pagination. I know that Komga has support with Cantook but that is for 2.0. 

While I don't know of any one app that supports both iOS and Android, there are many Android apps that do properly support OPDS 1.2 and iOS ones as well.
